tests/ElmSyntaxHelpersTest.elm

```diff
@@ -0,0 +1,80 @@
+module ElmSyntaxHelpersTest exposing (..)
+
+import Elm.Syntax.Node as Node exposing (Node(..))
+import Elm.Syntax.TypeAnnotation exposing (TypeAnnotation(..))
+import ElmSyntaxHelpers
+import Expect
+import Fuzz exposing (Fuzzer)
+import Test exposing (Test, describe, fuzz, fuzz2, test)
+
+
+typeAnnotationsMatchTests : Test
+typeAnnotationsMatchTests =
+    describe "typeAnnotationsMatch"
+        [ test "ranges are ignored" <|
+            \_ ->
+                let
+                    noRangeUnit =
+                        Node.empty Unit
+
+                    someRangeUnit =
+                        Node { start = { row = 5, column = 1 }, end = { row = 5, column = 16 } } Unit
+                in
+                ElmSyntaxHelpers.typeAnnotationsMatch noRangeUnit someRangeUnit
+                    |> Expect.equal True
+        , fuzz typeAnnotationsFuzzer "a TypeAnnotation matches itself" <|
+            \a ->
+                ElmSyntaxHelpers.typeAnnotationsMatch a a
+                    |> Expect.equal True
+        , fuzz2 typeAnnotationsFuzzer typeAnnotationsFuzzer "only matches equal TypeAnnotations" <|
+            \a b ->
+                ElmSyntaxHelpers.typeAnnotationsMatch a b
+                    |> Expect.equal (a == b)
```

Comment:
This silly looking fuzz test showed me a bug straight away 💛 
(I was comparing lists with `List.map2` and forgot to check that they needed to be of the same size)